File: app/src/main/java/afkt/project/ui/adapter/LinearSnapAdapter.java
package afkt.project.ui.adapter;

import androidx.annotation.Nullable;

import com.bumptech.glide.request.RequestOptions;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;

import java.util.List;

import afkt.project.R;
import afkt.project.model.bean.ItemBean;
import afkt.project.util.ProjectUtils;
import dev.other.GlideUtils;
import dev.utils.app.helper.ViewHelper;

/**
 * detail: RecyclerView Gallery 效果 Adapter
 * @author Ttt
 */
public class LinearSnapAdapter extends BaseQuickAdapter<ItemBean, BaseViewHolder> {

    RequestOptions roundOptions;

    public LinearSnapAdapter(@Nullable List<ItemBean> data) {
        super(R.layout.adapter_linear_snap, data);

        roundOptions = ProjectUtils.getRoundOptions10();
    }

    @Override
    protected void convert(
            BaseViewHolder helper,
            ItemBean item
    ) {
        ViewHelper.get()
                .setText(helper.getView(R.id.vid_als_title_tv), item.title)
                .setText(helper.getView(R.id.vid_als_subtitle_tv), item.subtitle)
                .setText(helper.getView(R.id.vid_als_time_tv), item.timeFormat);
        GlideUtils.with().displayImage(item.imageUrl, helper.getView(R.id.vid_als_igview),
                roundOptions);
    }
}

File: app/src/main/java/afkt/project/ui/adapter/LinearSnapAdapter.kt
package afkt.project.ui.adapter

import afkt.project.R
import afkt.project.databinding.AdapterLinearSnapBinding
import afkt.project.model.bean.ItemBean
import afkt.project.util.ProjectUtils
import android.view.ViewGroup
import dev.adapter.DevDataAdapter
import dev.base.adapter.DevBaseViewBindingVH
import dev.base.adapter.newBindingViewHolder
import dev.engine.DevEngine
import dev.utils.app.helper.view.ViewHelper

/**
 * detail: RecyclerView Gallery 效果 Adapter
 * @author Ttt
 */
class LinearSnapAdapter(data: List<ItemBean>) : DevDataAdapter<ItemBean, DevBaseViewBindingVH<AdapterLinearSnapBinding>>() {

    init {
        setDataList(data, false)
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): DevBaseViewBindingVH<AdapterLinearSnapBinding> {
        return newBindingViewHolder(parent, R.layout.adapter_linear_snap)
    }

    override fun onBindViewHolder(
        holder: DevBaseViewBindingVH<AdapterLinearSnapBinding>,
        position: Int
    ) {
        val item = getDataItem(position)
        ViewHelper.get()
            .setText(holder.binding.vidAlsTitleTv, item.title)
            .setText(holder.binding.vidAlsSubtitleTv, item.subtitle)
            .setText(holder.binding.vidAlsTimeTv, item.timeFormat)
        DevEngine.getImage()?.display(
            holder.binding.vidAlsIgview,
            item.imageUrl,
            ProjectUtils.roundConfig10
        )
    }
}